src/backend/Dimacs.ml

```diff
@@ -4,118 +4,135 @@ Copyright 2014 Guillaume Bury
 *)
 
+open Msat
+
+module type ARG = sig
+  type clause
+  val lits : clause -> int list
+end
+
 module type S = sig
+  type st
 
   type clause
+  (** The type of clauses *)
 
   val export :
+    st ->
     Format.formatter ->
     hyps:clause Vec.t ->
     history:clause Vec.t ->
     local:clause Vec.t ->
     unit
-  (** Export the given clause vectors to the dimacs format.
-      The arguments should be transmitted directly from the corresponding
-      function of the {Internal} module. *)
 
   val export_icnf :
     Format.formatter ->
     hyps:clause Vec.t ->
     history:clause Vec.t ->
     local:clause Vec.t ->
     unit
-  (** Export the given clause vectors to the dimacs format.
-      The arguments should be transmitted directly from the corresponding
-      function of the {Internal} module. *)
 
 end
 
-module Make(St : Solver_types.S)(Dummy: sig end) = struct
+module Make(St : Msat.S)(A: ARG with type clause = St.clause) = struct
+  type st = St.t
+
+  let pp_dimacs fmt c =
+    let lits = A.lits c in
+    List.iter (fun p -> Format.fprintf fmt "%d " p) lits;
+    Format.fprintf fmt "0"
+
 
   (* Dimacs & iCNF export *)
   let export_vec name fmt vec =
-    Format.fprintf fmt "c %s@,%a@," name (Vec.print ~sep:"" St.pp_dimacs) vec
+    Format.fprintf fmt "c %s@,%a@," name (Vec.pp ~sep:"" pp_dimacs) vec
 
   let export_assumption fmt vec =
-    Format.fprintf fmt "c Local assumptions@,a %a@," St.pp_dimacs vec
+    Format.fprintf fmt "c Local assumptions@,a %a@," pp_dimacs vec
 
   let export_icnf_aux r name map_filter fmt vec =
-    let aux fmt v =
+    let aux fmt _ =
       for i = !r to (Vec.size vec) - 1 do
         let x = Vec.get vec i in
         match map_filter x with
         | None -> ()
-        | Some y -> Format.fprintf fmt "%a@," St.pp_dimacs (Vec.get vec i)
+        | Some _ -> Format.fprintf fmt "%a@," pp_dimacs (Vec.get vec i)
       done;
       r := Vec.size vec
     in
     Format.fprintf fmt "c %s@,%a" name aux vec
 
+  (* TODO
   let map_filter_learnt c =
-    match c.St.cpremise with
+    match P.Clause.premise c with
     | St.Hyp | St.Local -> assert false
     | St.Lemma _ -> Some c
     | St.History l ->
       begin match l with
         | [] -> assert false
         | d :: _ ->
-          begin match d.St.cpremise with
+          begin match St.Clause.premise d with
             | St.Lemma _ -> Some d
             | St.Hyp | St.Local | St.History _ -> None
           end
       end
 
   let filter_vec learnt =
-    let lemmas = Vec.make (Vec.size learnt) St.dummy_clause in
+    let lemmas = Vec.create() in
     Vec.iter (fun c ->
         match map_filter_learnt c with
         | None -> ()
         | Some d -> Vec.push lemmas d
       ) learnt;
     lemmas
+     *)
 
-  let export fmt ~hyps ~history ~local =
-    assert (Vec.for_all (function
-        | { St.cpremise = St.Hyp; _} -> true | _ -> false
-      ) hyps);
+  let export st fmt ~hyps ~history ~local =
+    assert false
+    (* FIXME
```

Comment:
Why is this commented out ? I mean, did you encounter some example that made the assertions fail ?